ocaml/idl/datamodel_types.ml

```diff
@@ -67,159 +67,190 @@ type api_release = {
+     release_date  = Some "December 2010";
    }; {
      code_name     = Some rel_boston;
      version_major = 1;
      version_minor = 9;
-     branding   = "XenServer 6.0";
+     branding      = "XenServer 6.0";
+     release_date  = Some "September 2011";
    }; {
      code_name     = Some rel_tampa;
      version_major = 1;
      version_minor = 10;
-     branding   = "XenServer 6.1";
+     branding      = "XenServer 6.1";
+     release_date  = Some "September 2012";
    }; {
      code_name     = Some rel_clearwater;
      version_major = 2;
      version_minor = 0;
-     branding   = "XenServer 6.2";
+     branding      = "XenServer 6.2";
+     release_date   = Some "June 2013";
    }; {
      code_name     = Some rel_vgpu_tech_preview;
      version_major = 2;
      version_minor = 0;
-     branding   = "XenServer 6.2 SP1 Tech-Preview";
+     branding      = "XenServer 6.2 SP1 Tech-Preview";
+     release_date  = None;
    }; {
      code_name     = Some rel_vgpu_productisation;
      version_major = 2;
      version_minor = 1;
-     branding   = "XenServer 6.2 SP1";
+     branding      = "XenServer 6.2 SP1";
+     release_date = Some "December 2013";
    }; {
      code_name     = Some rel_clearwater_felton;
      version_major = 2;
      version_minor = 2;
-     branding   = "XenServer 6.2 SP1 Hotfix 4";
+     branding      = "XenServer 6.2 SP1 Hotfix 4";
+     release_date  = Some "April 2014";
    }; {
      code_name     = Some rel_clearwater_whetstone;
      version_major = 2;
      version_minor = 2;
-     branding   = "XenServer 6.2 SP1 Hotfix 11";
+     branding      = "XenServer 6.2 SP1 Hotfix 11";
+     release_date  = Some "October 2014";
    }; {
      code_name     = Some rel_creedence;
      version_major = 2;
      version_minor = 3;
-     branding   = "XenServer 6.5";
+     branding      = "XenServer 6.5";
+     release_date  = Some "January 2015";
    }; {
      code_name     = Some rel_cream;
      version_major = 2;
      version_minor = 4;
-     branding   = "XenServer 6.5 SP1";
+     branding      = "XenServer 6.5 SP1";
+     release_date  = Some "May 2015";
    }; {
      code_name     = Some rel_indigo;
      version_major = 2;
      version_minor = 4;
-     branding   = "XenServer 6.5 SP1 Hotfix 31";
+     branding      = "XenServer 6.5 SP1 Hotfix 31";
+     release_date  = None;
    }; {
      code_name     = Some rel_dundee;
      version_major = 2;
      version_minor = 5;
-     branding   = "XenServer 7.0";
+     branding      = "XenServer 7.0";
+     release_date  = Some "May 2016";
    }; {
      code_name     = Some rel_ely;
      version_major = 2;
      version_minor = 6;
-     branding   = "XenServer 7.1";
+     branding      = "XenServer 7.1";
+     release_date  = Some "February 2017";
    }; {
      code_name     = Some rel_honolulu;
      version_major = 2;
      version_minor = 6;
-     branding   = "XenServer 7.1 CU1";
+     branding      = "XenServer 7.1 CU1";
+     release_date  = None;
```

Comment:
I don't see why we wouldn't want the CUs to show if they contain publicly visible changes. I tend to think that the lack of certain metadata is only a result of negligence because until recently all these pieces of info were maintained manually and in more than one place. Perhaps @robhoes may be able to confirm this in case I'm missing something.

Reply:
Let's just fill in the release data for Honolulu.